pages/admin/dashboard/modules.tsx

```diff
@@ -69,23 +70,65 @@ const ModulesHeader: React.FC = () => {
 };
 
 const Modules: React.FC = () => {
+    const { isOpen, onOpen, onClose } = useDisclosure();
+    const [selectedModule, setSelectedModule] = useState("");
+    const [selectedModuleId, setSelectedModuleId] = useState("");
+    const [selectedModuleTool, setSelectedModuleTool] = useState("");
+    const deleteConfirmation =
+        selectedModuleTool === ""
+            ? `Are you sure you want to delete the ${selectedModule} module? This action cannot be undone.`
+            : `Are you sure you want to delete the ${selectedModule} module? This will unpublish the ${selectedModuleTool} tool and the ${selectedModule} module and cannot be undone.`;
```

Comment:
I don't actually unpublish the tool or the module here, since I'm not sure how to test it. I would appreciate any tips!

Reply:
Yoyoooo! So you have a reference, I implemented unpublishing and linking tools to modules in my new PR [here](https://github.com/uwblueprint/headsupguys/pull/82) feel free to take a look and implement it if you think it's good, and let me know if I could do some things better!

Reply:
But other than that, absolutely great job! Things look good. I'm pretty sure unpublishing and linking might be _technically_ out of this task scope, so if you don't want to implement them please let me know and I'll approve, otherwise feel free to ask me any questions regarding my implementation of unlinking and unpublishing.

Reply:
Thanks Chinemerem! I'll take a look tomorrow, and try impementating unlinking and unpublishing.

Reply:
Sweet if you have any problems please let me know!

Reply:
Hi Chinemerem, your implementation looks great! However, I do agree that it's a big change to be adding to this pr. Would it be ok to leave out unpublishing and linking for now (and add it later in another pr)?

Reply:
I think I'll assign this part to Chinemerem actually.